tests/test_free_only_mode.py

```
@@ -0,0 +1,190 @@
+
+    def test_free_only_mode_is_true(self, router):
+        """Verify free_only_mode is enabled in the router."""
+        assert router.free_only_mode is True, "free_only_mode should be True by default"
+
+    @pytest.mark.asyncio
+    async def test_forbidden_providers_not_in_candidates(self, router):
+        """Verify forbidden providers are not included in candidates."""
+        # Get candidates for a model that might have OpenAI configured
+        # The router should filter out forbidden providers
+
+        requirements = router._extract_requirements({})
```

Ruff (0.15.2)

[error] Local variable `requirements` is assigned to but never used

Remove assignment to unused variable `requirements`

(F841)